pkg/controller/bundles/bundles_syncer.go

```diff
@@ -48,40 +74,78 @@ func (syncer *LeafHubBundlesSpecSync) Start(stopChannel <-chan struct{}) error {
 	ctx, cancelContext := context.WithCancel(context.Background())
 	defer cancelContext()
 
-	go syncer.sync(ctx)
+	// start workers
+	for i := 0; i < len(syncer.k8sClients); i++ {
+		go syncer.runClientWorker(ctx, syncer.k8sClients[i])
+	}
 
-	for {
-		<-stopChannel // blocking wait for stop event
-		syncer.log.Info("stopped bundles syncer")
-		cancelContext()
+	go syncer.sync()
```

Comment:
This function should receive the context and handle cancellation.

Reply:
done